packages/contracts/test/voting/multisig.ts

```diff
@@ -338,140 +434,189 @@ describe('Multisig', function () {
 
-      // check for the `ProposalExecuted` event in the voting contract
-      {
-        const event = await findEvent(tx, VOTING_EVENTS.PROPOSAL_EXECUTED);
-        expect(event.args.proposalId).to.equal(id);
-      }
+        expect((await multisig.getProposal(id)).executed).to.be.true;
 
-      // calling execute again should fail
-      await expect(multisig.execute(id)).to.be.revertedWith(
-        customError('ProposalExecutionForbidden', id)
-      );
-    });
-  });
+        expect(await multisig.canExecute(id)).to.be.false;
+      });
+
+      it('returns `true` if the proposal can be executed', async () => {
+        await multisig.connect(signers[0]).approve(id, false);
+        await multisig.connect(signers[1]).approve(id, false);
+        await multisig.connect(signers[2]).approve(id, false);
 
-  describe('MultisigSettings', async () => {
-    it('should set the right minApprovals during initialization', async () => {
-      await multisig.initialize(dao.address, addresslist(5), {
-        onlyListed,
-        minApprovals,
+        expect((await multisig.getProposal(id)).executed).to.be.false;
+
+        expect(await multisig.canExecute(id)).to.be.true;
       });
-      expect((await multisig.multisigSettings()).minApprovals).to.be.eq(
-        minApprovals
-      );
     });
 
-    it('should set the right onlyListed during initialization', async () => {
-      await multisig.initialize(dao.address, addresslist(5), {
-        onlyListed,
-        minApprovals,
+    describe('execute:', async () => {
+      it('executes if the minimum approval is met', async () => {
+        await multisig.connect(signers[0]).approve(id, false);
+        await multisig.connect(signers[1]).approve(id, false);
+        await multisig.connect(signers[2]).approve(id, false);
+
+        const proposal = await multisig.getProposal(id);
+
+        expect(proposal.parameters.minApprovals).to.equal(
+          multisigSettings.minApprovals
+        );
+        expect(await multisig.approvals(id)).to.be.eq(
+          multisigSettings.minApprovals
+        );
+
+        expect(await multisig.canExecute(id)).to.be.true;
+        await expect(multisig.execute(id)).to.not.be.reverted;
       });
-      expect((await multisig.multisigSettings()).onlyListed).to.be.eq(
-        onlyListed
-      );
-    });
 
-    it('should emit MultisigSettingsUpdated during initialization', async () => {
-      await expect(
-        multisig.initialize(dao.address, addresslist(5), {
-          onlyListed,
-          minApprovals,
-        })
-      )
-        .to.emit(multisig, MULTISIG_EVENTS.MULTISIG_SETTINGS_UPDATED)
-        .withArgs(onlyListed, minApprovals);
-    });
+      it('executes if the minimum approval is met and can be called by an unlisted accounts', async () => {
+        await multisig.connect(signers[0]).approve(id, false);
+        await multisig.connect(signers[1]).approve(id, false);
+        await multisig.connect(signers[2]).approve(id, false);
```

Comment:
I would also check that execution attempts fail before the actual approvals

Reply:
Added a test to check this separately f6ef2ab2e12930b80758c64f0ac107b6c1248a0d